### PR TITLE
PTS UUID generation

### PR DESCRIPTION
**Pull request by ilakyapal**
Add basic logic to connect permits mentioned in the "related_building_permit" field in PPTS with their correct project UUID. 

(I need to add tests to make sure records are properly connected but figured I'd get this out for now for comment)

### CODE REVIEW

File: schemaless/create_uuid_map.py [hunk 1]
```diff
@@ -17,25 +18,45 @@ def __init__(self):
...
+        latest_records = latest_values(schemaless_file)
+
+        # Create a mapping between permit numbers and their associated PPTS
+        # record (so we an ensure they are assigned the same UUID).
+        permit_number_to_ppts = {}
+        for fk, record in latest_records[PPTS.NAME].items():
+            if record['related_building_permits']:
+                for permit_number in record['related_building_permit'].split(","):
+                    #  This assumes that a given permit can only have one PPTS
+                    #  parent. Is that accurate?
```
Comment (sbuss) [inline schemaless/create_uuid_map.py]: I suspect that this isn't true

File: schemaless/create_uuid_map.py [hunk 2]
```diff
@@ -17,25 +18,45 @@ def __init__(self):
...
+                children = []
+                if source == PPTS.NAME:
+                    if record['parents']:
+                        parents = record['parents'].split(",")
+                    if record['children']:
+                        children = record['children'].split(",")
+
+                if source == PTS.NAME:
+                    if permit_number_to_ppts[record['permit_number']]:
+                      parents = [permit_number_to_ppts[record['permit_number']]]
```
Comment (sbuss) [inline schemaless/create_uuid_map.py]: another 2-space tab

File: schemaless/create_schemaless.py [hunk 3]
```diff
@@ -42,11 +42,11 @@ def just_dump(sources, outfile, the_date=None):
 
 def latest_values(schemaless_file):
     """Collapse the schemaless file into the latest values for each record."""
-    records = defaultdict(lambda: defaultdict(str))
+    records = defaultdict(lambda: defaultdict(lambda: defaultdict(str)))
     with open(schemaless_file, 'r') as inf:
         reader = DictReader(inf)
         for line in reader:
-            records[line['fk']][line['name']] = line['value']
+          records[line['source']][line['fk']][line['name']] = line['value']
```
Comment (sbuss) [inline schemaless/create_schemaless.py]: 2-space tab should be 4 spaces

File: schemaless/create_uuid_map.py (hunk 1 above)
Comment (sbuss) [reply schemaless/create_uuid_map.py]: easily solvable with a `defaultdict(list)` and `permit_number_to_ppts[permit_number].append(fk)`

File: schemaless/create_uuid_map.py [hunk 4]
```diff
@@ -17,25 +18,45 @@ def __init__(self):
...
-                date=the_date,
-                parents=parents,
-                children=children,
-                uuid=None,
-            ))
+        for source, source_records in latest_records.items():
+            for fk, record in source_records.items():
+                parents = []
+                children = []
+                if source == PPTS.NAME:
```
Comment (sbuss) [inline schemaless/create_uuid_map.py]: I'm going to try moving this logic to the `Source` objects when I do the same thing for TCO data

Comment (ilakyapal) [reply schemaless/create_uuid_map.py]: sgtm, thanks!

File: schemaless/create_uuid_map.py [hunk 5]
```diff
@@ -17,25 +19,46 @@ def __init__(self):
...
+                    if record['parents']:
+                        parents = record['parents'].split(",")
+                    if record['children']:
+                        children = record['children'].split(",")
+
+                if source == PTS.NAME:
+                    if permit_number_to_ppts[record['permit_number']]:
+                        parents = [permit_number_to_ppts[
+                            record['permit_number']]]
```
Comment (rajpara11) [inline schemaless/create_uuid_map.py]: Sorry stupid question since I'm trying to drive-by understand the code, how does this handle the permit number itself not being the fk, but the record id being the foreign key?

Comment (ilakyapal) [reply schemaless/create_uuid_map.py]: If you look at the logic below where the Node actually gets added, things are keyed with the correct fk (which in the case of PTS will be record_id, as per the sources.py file). 

The permit_number_to_ppts dict is just something I computed above by iterating through PPTS records and looking at 'RELATED_BUILDING_PERMIT' for the purposes of intermediate computation -- it maps the list of PPTS fks that would be the "parent" for a given PTS record with a given permit number, since that is what people input in PPTS instead of the actual pts record id (tangential thought, right now some of the field naming 
for PPTS is a bit confusing -- if you look above, 'building_permit_id' is actually the field name that we mapped to 'RELATED_BUILDING_PERMIT', which is actually usually a permit number... I'll probably rename this in follow up).

**Comment by ilakyapal** [comment]
@sbuss Running gen-test-data.sh currently does not produce any changes I think since we don't have pts test data in there yet. I'll add that in follow up PR

File: schemaless/create_uuid_map.py (hunk 5 above)
Comment (rajpara11) [reply schemaless/create_uuid_map.py]: Ahh gotcha, thanks for the explanation, i think i misunderstood some of the field names but the code makes sense now

**Comment by ilakyapal** [comment]
Fixes https://github.com/sfcpc/housing-dashboard/issues/21
